### ISSUE TITLE
MDEV-14767 system_versioning_alter_history should be included into binlog events

### ISSUE THREAD

**Issue by midenok** [tempesta-tech/mariadb#466]
### Reproduce
```sql
--source include/master-slave.inc
--
 
create or replace table t1 (a int) with system versioning;
set system_versioning_alter_history= KEEP;
alter table t1 add column b int;
--sync_slave_with_master
 
# Cleanup
--connection master
drop table t1;
--source include/rpl_end.inc
```
### Result
```sql
Last_Error	Error 'Not allowed for system-versioned `test`.`t1`. Change @@system_versioning_alter_history to proceed with ALTER.' on query. Default database: 'test'. Query: 'alter table t1 add column b int'
```
### Expected
No error.
### Info
Related to #94, #234.

**Comment by midenok** [comment]
In **log_event.cc** see this:
```c++
   5132     if (unlikely(tmp)) /* some bits have changed */
   5133     {
   5134       bool need_comma= 0;
   5135       if (my_b_write_string(file, "SET ") ||
   5136           print_set_option(file, tmp, OPTION_NO_FOREIGN_KEY_CHECKS, ~flags2,
   5137                            "@@session.foreign_key_checks", &need_comma)||
   5138           print_set_option(file, tmp, OPTION_AUTO_IS_NULL, flags2,
   5139                            "@@session.sql_auto_is_null", &need_comma) ||
   5140           print_set_option(file, tmp, OPTION_RELAXED_UNIQUE_CHECKS, ~flags2,
   5141                            "@@session.unique_checks", &need_comma) ||
   5142           print_set_option(file, tmp, OPTION_NOT_AUTOCOMMIT, ~flags2,
   5143                            "@@session.autocommit", &need_comma) ||
   5144           print_set_option(file, tmp, OPTION_NO_CHECK_CONSTRAINT_CHECKS,
   5145                            ~flags2,
   5146                            "@@session.check_constraint_checks", &need_comma) ||
   5147           my_b_printf(file,"%s\n", print_event_info->delimiter))
   5148         goto err;
   5149       print_event_info->flags2= flags2;
   5150     }
```
and this:
```sql
   5166   if (likely(sql_mode_inited) &&
   5167       (unlikely(print_event_info->sql_mode != sql_mode ||
   5168                 !print_event_info->sql_mode_inited)))
   5169   {
   5170     char llbuff[22];
   5171     if (my_b_printf(file,"SET @@session.sql_mode=%s%s\n",
   5172                     ullstr(sql_mode, llbuff), print_event_info->delimiter))
   5173       goto err;
   5174     print_event_info->sql_mode= sql_mode;
   5175     print_event_info->sql_mode_inited= 1;
   5176   }
```
etc.

**Comment by FooBarrior** [comment]
I'd rather write the variable each time alter table occurs instead of including it in the log header

**Comment by midenok** [comment]
Fixed by dd7d169593f70fd6894ffadc5ce99bf9cf4907a2
